client/src/pages/App/App.tsx

```diff
@@ -14,25 +14,31 @@ import LoginPage from "../Login/LoginPage";
 import MatesPage from "../Mates/MatesPage";
 import NewMatesPage from "../NewMates/NewMatesPage";
 
-const App = (): JSX.Element => {
+/**
+ * Main App page
+ * @returns JSX.Element or null
+ */
+const App = (): JSX.Element | null => {
   const [loggedIn, setLoggedIn] = useState(false);
+  const [checkedLoggedIn, setCheckedLoggedIn] = useState(false);
```

Comment:
Would this be better implemented as an enum: loggedIn, notLoggedIn, waiting.

Reply:
it's a bit different, but it could work